Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -3010,6 +3012,24 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     compileDeclarativeNetRequestRules(allJSONData, WTFMove(completionHandler));
 }
 
+
+void WebExtensionContext::handleContentRuleListNotificationForTab(WebExtensionTab& tab, const URL& url, WebCore::ContentRuleListResults::Result)
+{
+    incrementActionCountForTab(tab, 1);
+
+    if (!hasPermission(_WKWebExtensionPermissionDeclarativeNetRequestFeedback) && !(hasPermission(_WKWebExtensionPermissionDeclarativeNetRequest) && hasPermission(_WKWebExtensionPermissionActiveTab)))
+        return;
+
+    // FIXME: Set a timer to purge old matched rules.
```

Comment:
See `WebExtensionAlarm.cpp` for a simple-ish timer example.

Reply:
I'll put this into the next patch, and add a bug for the FIXME in the meantime.